src/BasicLogAnalyser.php

```diff
@@ -290,8 +297,46 @@ private function parseLogEntry($line) {
 			//don't have their own log columns. Extract them from the query string.
 			$result['php_version'] = null;
 			$result['locale'] = null;
+
+
 			if (!empty($result['query_string'])) {
 				parse_str($result['query_string'], $parameters);
+				
+				if (isset($parameters['cms'])) {
+					$parameters['cms'] = str_replace('=', '', $parameters['cms']);
+					
+					
+					if (strpos($parameters['cms'], 'ClassicPress') !== false) {
+						$result['cms'] = 'CP';
+
+						/**
+						 *
+						 * If ClassicPress is detected, will override the cms version with the param "ver" of the site_url
```

Comment:
I don't think I fully understand this whole section. Why and how does the site_url include the ClassicPress version number? Is this a default CP feature or something that you added? If you could post an example URL or two in a comment that would help.

Reply:
Hi, 

This is a feature of CP, to ensure compatibility with third-party services that expect requests to come from WordPress.

This are real entries from request.log, just hide the IP, SLUG, DOMAIN and TOKEN

[2019-04-12 11:20:42 -0500] IP  	GET 	get_metadata	SLUG	1.0.18	4.9.10	https://www.DOMAIN.net/?wp_compatible=true&ver=1.0.1	action=get_metadata&slug=padma-services&token=TOKEN&installed_version=1.0.18&php=7.0.33-6+ubuntu16.04.1+deb.sury.org+3&locale=en_US&checking_for_updates=1

[2019-04-24 18:19:27 -0500] IP    	GET 	get_metadata	SLUG	1.0.4	4.9.10	https://www.DOMAIN.net/?wp_compatible=true&ver=1.0.1	action=get_metadata&slug=padma-updater&token=TOKEN&installed_version=1.0.4&php=7.0.33-6+ubuntu16.04.1+deb.sury.org+3&locale=en_US&checking_for_updates=1


This is from ClassicPress wp-includes/http.php:

/**
 * Returns the ClassicPress User-Agent header for outgoing requests.
 *
 * In WordPress, this header looks like "WordPress/X.Y.Z; https://siteurl.com".
 * To ensure compatibility with third-party services that expect requests to
 * come from WordPress, ClassicPress uses the same pattern, but uses the URL to
 * indicate that requests come from a ClassicPress site.
 *
 * @since 1.0.0
 *
 * @param bool $include_site_id Whether to include a site identifier (used when
 *                              communicating with the ClassicPress API).  This
 *                              is not personally identifiable information.
 * @return string The User-Agent header value.
 */
function classicpress_user_agent( $include_site_id = false ) {
	$url = 'https://www.classicpress.net/?wp_compatible=true'
		. '&ver=' . classicpress_version_short();

	if ( $include_site_id ) {
		$url .= '&site=' . sha1( preg_replace(
			'#^https?:#',
			'',
			strtolower( untrailingslashit( home_url( '/' ) ) )
		) );
	}

	$user_agent = 'WordPress/' . get_bloginfo( 'version' ) . '; ' . $url;

	/**
	 * Filters the user agent value sent with an HTTP request.
	 *
	 * @since WP-2.7.0
	 * @since 1.0.0 This filter is applied to all HTTP requests.
	 *
	 * @param string $user_agent ClassicPress user agent string.
	 */
	return apply_filters( 'http_headers_useragent', $user_agent );
}